src/web/mjs/connectors/templates/SinMH.mjs

```diff
@@ -83,7 +84,7 @@ export default class SinMH extends Connector {
             });
         `;
         let request = new Request(this.url + chapter.id, this.requestOptions);
-        let data = await Engine.Request.fetchUI(request, script);
+        let data = await Engine.Request.fetchUI(request, this.scriptPages || script);
```

Comment:
Remove local variable `script` and assign its value to `this.scriptPages`

Reply:
i tried that, couldn't get it to work, i assumed it had something to with it having ${this.api} in the script

Reply:
nvm i guess i did something else wrong whilst i was doing it

Reply:
if `${this.api}` (or in the chapter script `${this.queryChapters}`) is changed by the connector then with the script in this.scriptPages it won't get the changed value and will instead get the template value

Reply:
unless this is sorted it can't be moved 
if it is sorted then manhuagui chapters can taken out too

Reply:
Then leave as is for now